perun-ldapc-ada/src/main/java/cz/metacentrum/perun/ldapc/model/impl/PerunVOImpl.java

```diff
@@ -144,4 +145,25 @@ public Name getEntryDN(String ...voId) {
 				.build();
 	}
 
+	@Override
+	public List<Name> listEntries() throws InternalErrorException {
+		return ldapTemplate.search(query().
+				where("objectclass").is(PerunAttribute.PerunAttributeNames.objectClassPerunVO),
+				getNameMapper());
+	}
+
+	@Override
+	public void deleteEntry(Name dn) throws InternalErrorException {
+		// first find and remove entries in subtree
+		List<Name> subentries = ldapTemplate.search(query()
```

Comment:
This search returns also base itself, which can't be deleted. Either use different ``search()`` method with specified SearchScope or simply check returned ``Name`` on orignal ``dn`` param and skip it in the cycle below.